mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/impl/PojoMassIndexingBatchCoordinator.java

```diff
@@ -83,6 +84,10 @@ public PojoMassIndexingBatchCoordinator(PojoMassIndexingMappingContext mappingCo
 				mappingContext.threadPoolProvider(),
 				mappingContext.failureHandler()
 		);
+
+		if ( failFast ) {
+			getNotifier().addOnFailureHandlerCallback( this::cleanUpOnFailure );
```

Comment:
Hey  @yrodiere 

The idea for this fail-fast implementation is to cancel the coordinator once the failure handler is called. Cancelling the coordinator also cancels the sub-tasks that the coordinator started. So that seems like a simple solution....
Do you think this approach makes sense, or maybe you'd have some other suggestion? Thanks! 😃

Reply:
mmm a fair point 😃 
I guess... I wanted it to be more explicit, i.e. call the cancel because we were in the "fail-fast-if-statement" 🙈.

With the boolean and an exception, this seems to work ok (throwing a `MassIndexingOperationHandledFailureException` so that we do not need to add anything to the exception handling)